Realm/RLMSyncSession.h

```diff
@@ -56,6 +102,32 @@ NS_ASSUME_NONNULL_BEGIN
 /// session.
 - (nullable RLMSyncConfiguration *)configuration;
 
-NS_ASSUME_NONNULL_END
+/**
+ Register a progress notification block. Multiple blocks can be registered on the same session at once. All blocks
+ will be dispatched to the main queue.
```

Comment:
Dispatching notifications to a queue rather than the thread the notification block was added on is inconsistent with all of our other notifications.